spec/concerns/queue_it/queable_spec.rb

```diff
@@ -148,6 +148,146 @@
     end
   end
 
+  describe '#get_next_node_in_queue_by' do
+    context "when queue dosen't exist or is empty" do
+      it { expect(task.get_next_node_in_queue_by('id', 1)).to be(nil) }
+    end
+
+    context 'when queue has one node' do
+      let(:nodable) { create(:user) }
+
+      before { task.push_to_queue(nodable) }
+
+      it "expects to find head_node and do not change it's kind value" do
+        expect(task.get_next_node_in_queue_by('id', nodable.id).nodable).to eq(nodable)
+        expect(task.queue.nodes.find_by(kind: :head).nodable).to eq(nodable)
+      end
+    end
+
+    context 'when queue has two nodes' do
+      let!(:first_nodable) { create(:user) }
+      let!(:second_nodable) { create(:user) }
+      let(:in_head) { true }
+
+      before do
+        task.push_to_queue(first_nodable)
+        task.push_to_queue(second_nodable, in_head)
+      end
+
+      context 'when searching for tail node' do
+        it "expects to return first_nodable and to not change nodables kind's" do
+          expect(task.get_next_node_in_queue_by('id', first_nodable.id).nodable).to eq(first_nodable)
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/LineLength: Line is too long. [101/100]